src/main/java/com/openlattice/chronicle/controllers/v2/ChronicleControllerV2.java

```diff
@@ -0,0 +1,391 @@
+            produces = { MediaType.APPLICATION_JSON_VALUE, CustomMediaType.TEXT_CSV_VALUE }
+    )
+    public Iterable<Map<String, Set<Object>>> getAllParticipantAppsUsageData(
+            @PathVariable( ORGANIZATION_ID ) UUID organizationId,
+            @PathVariable( STUDY_ID ) UUID studyId,
+            @PathVariable( ENTITY_KEY_ID ) UUID participantEntityKeyId,
+            @RequestParam( value = FILE_TYPE, required = false ) FileType fileType,
+            HttpServletResponse response ) {
+
+        Iterable<Map<String, Set<Object>>> data = getAllParticipantAppsUsageData(
+                organizationId,
+                studyId,
+                participantEntityKeyId,
+                fileType );
+
+        String fileName = getParticipantDataFileName(
+                enrollmentManager,
+                USAGE_DATA_PREFIX,
+                organizationId,
+                studyId,
+                participantEntityKeyId );
+        setContentDisposition( response, fileName, fileType );
+        setDownloadContentType( response, fileType );
+        return data;
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = ORGANIZATION_ID_PATH + STUDY_ID_PATH + QUESTIONNAIRES_PATH,
+            method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Map<UUID, Map<FullQualifiedName, Set<Object>>> getStudyQuestionnaires(
+            @PathVariable( ORGANIZATION_ID ) UUID organizationId,
+            @PathVariable( STUDY_ID ) UUID studyId ) {
+
+        return surveysManager.getStudyQuestionnaires( organizationId, studyId );
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = ORGANIZATION_ID_PATH + STUDY_ID_PATH + PARTICIPANT_ID_PATH + DATASOURCE_ID_PATH + UPLOAD_PATH,
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE )
+    public Integer upload(
+            @PathVariable( ORGANIZATION_ID ) UUID organizationId,
+            @PathVariable( STUDY_ID ) UUID studyId,
+            @PathVariable( PARTICIPANT_ID ) String participantId,
+            @PathVariable( DATASOURCE_ID ) String datasourceId,
+            @RequestBody List<SetMultimap<UUID, Object>> data ) {
+
+        return dataUploadManager.logData( organizationId, studyId, participantId, datasourceId, data );
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = EDM_PATH,
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public Map<String, UUID> getPropertyTypeIds( Set<String> propertyTypeFqns ) {
+        return enrollmentManager.getPropertyTypeIds( propertyTypeFqns );
+    }
+
+    @Override
+    @Timed
+    @RequestMapping(
+            path = STATUS_PATH,
+            method = RequestMethod.GET
+    )
+    public Boolean isRunning() {
+
+        //TODO: Ensure connectivity with OpenLattice backend.
```

Comment:
No, this todo is a long-existing carry-over from the existing api. I reckon it would be good to implement it at some point, but not necessarily right away